pandas/tests/io/formats/test_to_html.py

```diff
@@ -415,6 +431,85 @@ def test_to_html_multiindex_max_cols(self, datapath):
+        df.columns = MultiIndex.from_product([['a', 'b'], ['c', 'd']])
+        df.index = MultiIndex.from_product([['a', 'b'],
+                                            ['c', 'd', 'e', 'f', 'g']])
+        result = df.to_html(index=False)
+        expected = expected_html(datapath, 'gh22579_expected_output')
+        assert result == expected
+
+    @pytest.mark.parametrize(
+        ('idx_flag', 'named_index', 'expected_output'),
+        [(True, True, 'gh22579_truncation_index_true'),
+         (False, True, 'gh22579_truncation_index_false_named'),
+         (False, False, 'gh22579_truncation_index_false_unnamed')
+         ])
+    def test_to_html_multi_indexes_index_false_with_truncation(
+            self, datapath, idx_flag, named_index, expected_output):
+        # GH 22579 with truncation
+        names = ['foo', None, 'baz'] if named_index else None
+        index = MultiIndex.from_product([['a', 'b'], ['c', 'd'], ['e', 'f']],
+                                        names=names)
+        df = DataFrame(np.arange(64).reshape(8, 8), index=index, columns=index)
+        result = df.to_html(max_rows=4, max_cols=4, index=idx_flag)
+        expected = expected_html(datapath, expected_output)
+        assert result == expected
+
+    @pytest.mark.parametrize('col_idx_type',
+                             [Index([0, 1]),
+                              Index([0, 1], name='columns.name'),
+                              MultiIndex.from_product([['a'], ['b', 'c']]),
+                              MultiIndex.from_product([['a'], ['b', 'c']],
+                                                      names=['columns.name.0',
+                                                             'columns.name.1'])
+                              ],
+                             ids=['columns_unnamed_standard',
+                                  'columns_named_standard',
+                                  'columns_unnamed_multi',
+                                  'columns_named_multi'])
+    @pytest.mark.parametrize('idx_type',
+                             [Index([0, 1]),
+                              Index([0, 1], name='index.name'),
+                              MultiIndex.from_product([['a'], ['b', 'c']]),
+                              MultiIndex.from_product([['a'], ['b', 'c']],
+                                                      names=['index.name.0',
+                                                             'index.name.1'])
+                              ],
+                             ids=['index_unnamed_standard',
+                                  'index_named_standard',
+                                  'index_unnamed_multi',
+                                  'index_named_multi'])
+    def test_to_html_basic_alignment(self, datapath, idx_type, col_idx_type,
+                                     index, index_names):
+        # GH 22747, GH 22579
+        def _exp_name(idx_type, index, index_names):
```

Comment:
This is a rather complicated test. Can you not alternately inspect the ids of the fixture to get what you want instead?

Reply:
the only methods i've found on getting the `id` inside the test function all appear a bit hacky. the 'id' seems to be mainly to make the test output more readable.

i originally coded the index paramatrization as a fixture, but since it is only used for this test moved to a `parametrize` decorator. If i don't have different names for column and row indexes in the tests then a single fixture could be used for both the column and row indexes.

One way to simplify this is to use the the `id` as the parameter and then construct the index inside the test function. That would remove the need to introspect the index. The test would still need to construct the mapping from the test parameters to the expected output so would not remove the apparent complexity while making it more complicated to move the indexes into a fixture if they could be used in another test.

The mapping could be simplified by creating the map explicitly but this would result in a dict with  64 (4 x 4 x 2 x 2) loc. I am planning to add header parametrization to this test in a future PR to fix a couple of `header=False` issues. the explicit map would then be 128 loc.

I have tried all these methods at some point and this solution I believe is the simplest without contaminating the module namespace. Adding a `is_multiindex` and `has_names` helper function into the module namespace would make the code more readable.

```python
def _has_names(index):
    if index.nlevels > 1:
        return com._any_not_none(*index.names)
    else:
        return index.name is not None
```

Reply:
issue #23788 raised for the header=False issues which is why the header parameterisation has been removed from the basic alignment checks for this PR but will be added back in when this issue is addressed.

Reply:
If inspecting the id isn't an option couldn't you just pass in the name as an argument instead?